src/annotations.h

```diff
@@ -6,18 +6,15 @@
 #ifndef TIMESCALEDB_ANNOTATIONS_H
 #define TIMESCALEDB_ANNOTATIONS_H
 
-/* Fall-through annotation */
-#if defined(__clang__)
-#if (__clang_major__ >= 12) || (__clang_analyzer__)
+
+/* Supported since clang 12 and GCC 7 */
```

Comment:
we are not supporting older compiler versions?

Reply:
we are the fallthrough marking won't work on those but otherwise they are still supported

Reply:
We actually have some defines about it in the cmake file. For eg `CC_SUPPORTS_IMPLICIT_FALLTHROUGHT`, maybe rely on it if makes sense?

Reply:
this tests whether the compiler supports -Wimplicit-fallthrough not necessarily whether it supports fallthrough attribute